OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/optimobile/OptimovePreferenceCenter.java

```diff
@@ -0,0 +1,265 @@
+        }
+
+        @NonNull
+        public int getValue() {
+            return channel;
+        }
+
+        public static Channel getChannelByValue ( int value) {
+            switch (value) {
+                case 489:
+                    return MOBILE_PUSH;
+                case 490:
+                    return WEB_PUSH;
+                case 493:
+                    return SMS;
+                default:
+                    throw new IllegalArgumentException("Preference center does not support channel " + value);
+            }
+        }
+    }
+
+    public static OptimovePreferenceCenter getInstance() {
+        if (shared == null) {
+            throw new IllegalStateException("OptimovePreferenceCenter is not initialized");
+        }
+        return shared;
+    }
+
+    /**
+     * Asynchronously runs preferences get handler on UI thread. Handler receives a single argument Preferences
+     *
+     * @param preferencesGetHandler handler
+     */
+    public void getPreferencesAsync(@NonNull PreferencesGetHandler preferencesGetHandler) {
+        if (customerId == null) {
+            Optimobile.log(TAG, "No customer ID has been set");
+            return;
+        }
+
+        Runnable task = new GetPreferencesRunnable(preferencesGetHandler);
+        Optimobile.executorService.submit(task);
+    }
+
+    /**
+     *  Asynchronously runs preferences set handler on UI thread. Handler receives a single Boolean result argument
+     *
+     * @param preferencesSetHandler handler
+     * @param updates list of preference updates to set
+     */
+    public void setCustomerPreferencesAsync(@NonNull PreferencesSetHandler preferencesSetHandler, List<PreferenceUpdate> updates) {
+        if (customerId == null) {
+            Optimobile.log(TAG, "No customer ID has been set");
+            return;
+        }
+
+        Runnable task = new SetPreferencesRunnable(preferencesSetHandler, updates);
+        Optimobile.executorService.submit(task);
+    }
+
+    static class GetPreferencesRunnable implements Runnable {
+        private final PreferencesGetHandler callback;
+
+        GetPreferencesRunnable(PreferencesGetHandler callback) {
+            this.callback = callback;
+        }
+
+        @Override
+        public void run() {
+            String mappedRegion = shared.mapRegion(config.getRegion());
+            String url = "https://preference-center-" + mappedRegion + ".optimove.net/api/v1/preferences?customerId=" + customerId + "&brandGroupId=" + config.getBrandGroupId();
+
+            OptimobileHttpClient httpClient = OptimobileHttpClient.getInstance();
+
+            try (Response response = httpClient.getSync(url)) {
+                if (!response.isSuccessful()) {
+                    logFailedResponse(response);
+                } else {
+                   Preferences preferences = mapResponseToPreferences(response);
+
+                   if (preferences == null) {
+                       // log?
```

Comment:
log already when mapping response, prob can just remove this if completely -- still can fire callback